fact-bounty-client/src/pages/login/components/Login.js

```diff
@@ -91,8 +94,43 @@ class Login extends Component {
     this.setState(state => ({ showPassword: !state.showPassword }));
   };
   onChange = e => {
-    this.setState({ [e.target.id]: e.target.value });
+    let { id, value } = e.target;
+    this.setState({ [id]: value }, () => {
+      this.validateField(id, value);
+    });
   };
+
+  validateField = (fieldName, value) => {
+    let { emailValid, passwordValid, errors } = this.state;
+
+    switch (fieldName) {
+      case "email":
+        emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
+        errors.email = emailValid ? "" : "Invalid E-mail";
+        break;
+      case "password":
+        passwordValid = value.length >= 8;
+        errors.password = passwordValid ? "" : "Password too short!";
```

Comment:
I feel the error message `Password too short!` don't make sense in the login page, Instead, make a general error message which says `Wrong password. Try again or click Forgot password to reset it.`.